sympy/solvers/inequalities.py

```diff
@@ -1007,3 +1009,154 @@ def reduce_inequalities(inequalities, symbols=[]):
+                MM[ii, jj] = M[ii, jj] - M[ii, j] * M[i, jj] / M[i, j]
+    return MM
+
+
+def _simplex(M, R, S, random_seed=0):
+    """
+    Simplex method with randomized pivoting
+    http://web.tecnico.ulisboa.pt/mcasquilho/acad/or/ftp/FergusonUCLA_LP.pdf
+    """
+    rand = random.Random(x=random_seed)
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        C = M[-1, :-1]
+        if all(B[i] >= 0 for i in range(B.rows)):
+            piv_cols = []
+            for j in range(C.cols):
+                if C[j] < 0:
+                    piv_cols.append(j)
+
+            if not piv_cols:
+                return M, R, S
+            rand.shuffle(piv_cols)
+            j0 = piv_cols[0]
+
+            piv_rows = []
+            for i in range(A.rows):
+                if A[i, j0] > 0:
+                    ratio = B[i] / A[i, j0]
+                    piv_rows.append((ratio, i))
+
+            if not piv_rows:
+                raise UnboundedLinearProgrammingError('Objective function can assume arbitrarily large positive (resp. negative) values at feasible vectors!')
+            piv_rows = sorted(piv_rows, key=lambda x: (x[0], x[1]))
+            piv_rows = [(ratio, i) for ratio, i in piv_rows if ratio == piv_rows[0][0]]
+            rand.shuffle(piv_rows)
+            _, i0 = piv_rows[0]
+
+            M = _pivot(M, i0, j0)
+            R[j0], S[i0] = S[i0], R[j0]
+        else:
+            for k in range(B.rows):
+                if B[k] < 0:
+                    break
+
+            piv_cols = []
+            for j in range(A.cols):
+                if A[k, j] < 0:
+                    piv_cols.append(j)
+            if not piv_cols:
+                raise InfeasibleLinearProgrammingError('The constraint set is empty!')
+            rand.shuffle(piv_cols)
+            j0 = piv_cols[0]
+
+            ratio = B[k] / A[k, j0]
+            piv_rows = [(ratio, k)]
+            for i in range(A.rows):
+                if A[i, j0] > 0 and B[i] > 0:
+                    ratio = B[i] / A[i, j0]
+                    piv_rows.append((ratio, i))
+
+            piv_rows = sorted(piv_rows, key=lambda x: (x[0], x[1]))
+            piv_rows = [(ratio, i) for ratio, i in piv_rows if ratio == piv_rows[0][0]]
+            rand.shuffle(piv_rows)
+            _, i0 = piv_rows[0]
+
+            M = _pivot(M, i0, j0)
+            R[j0], S[i0] = S[i0], R[j0]
+
+
+# Maximize Cx + D constrained with Ax <= B and x >= 0
+# Minimize y^{T}B constrained with y^{T}A >= C^{T} and y >= 0
+def linear_programming(A, B, C, D = Matrix([0])):
+    """
+    When x is a column vector of variables, y is a column vector of dual variables,
+    and when the objective is either:
+
+    *) Maximizing Cx constrained to Ax <= B and x >= 0.
+    *) Minimizing y^{T}B constrained to y^{T}A >= C^{T} and y >= 0.
+
+    The method returns a triplet of solutions optimum, argmax and argmax_dual where
+    optimum is the optimum solution, argmax is x and argmax_dual is y.
```

Comment:
There should be explicit parameters and returns sections in the docstring:
https://docs.sympy.org/latest/documentation-style-guide.html#docstring-sections